src/source/test_waveform.rs

```diff
@@ -0,0 +1,196 @@
+//! Generator sources for various periodic test waveforms.
+//!
+//! This module provides several periodic, deterministic waveforms for testing other sources and
+//! for simple additive sound synthesis. Every source is monoaural and in the codomain `[-1.0f32,
```

Comment:
if its for testing *and* sound synthesis then neither *synth* or *test* are the right terms to describe it. Maybe waveform?